src/base/dof_map_constraints.C

```diff
@@ -431,6 +431,9 @@ private:
         std::vector<bool> is_boundary_node(elem->n_nodes(), false),
           is_boundary_edge(elem->n_edges(), false),
           is_boundary_side(elem->n_sides(), false);
+
+        // We also maintain a separate list of nodeset-based boundary nodes
+        std::vector<bool> is_boundary_nodeset(elem->n_nodes(), false);
```

Comment:
> "i++" on an integer.

The difference being that `i++` is a write operation, while *read-only* access into `vector<bool>` is thread-unsafe...